fix requirements and about details

bhazelton opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

bhazelton commented:
@conda-forge-admin, please rerender

bhazelton commented:
@conda-forge-admin, please rerender

ocefpaf commented:
@pkgw I think that @bhazelton in on the right path here but there is compilation error:

```
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
pyuvdata/src/miriad_wrap.cpp:4:25: fatal error: aipy_compat.h: No such file or directory
 #include "aipy_compat.h"
```

pkgw commented:
@ocefpaf Yeah, that's our busted tarball; we're rolling a 1.3.1 that ought to fix it.

bhazelton commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * The `requirements/` sections should be defined in the following order: build, host, run; instead saw: host, build, run.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

bhazelton commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I rerendered the feedstock and it seems to be already up-to-date.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

pkgw commented on recipe/meta.yaml:
```diff
@@ -11,43 +11,48 @@ source:
-  number: 0
-  script: python setup.py install --single-version-externally-managed --record=files.txt
-  skip: true  # [win or py3k]
+  number: 1
```
Since we're bumping the version number, the build number can be reset to 0.

bhazelton replied:
oops, thanks for catching that!

pkgw commented on recipe/meta.yaml:
```diff
@@ -11,43 +11,48 @@ source:
     - setuptools
+  host:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
```
Oh, I may have done this wrong in my commit. The requirements that aren't strictly build tools should go in the `host` stanza rather than the `build` stanza — that would be everything except the `compiler()` lines here, I think.

pkgw reviewed (commented):
One minor comment, one that might be important.

pkgw committed:
Update for 1.3.1

Deps change, preferred GitHub repo changes, and we are now compatible with Python 3.

pkgw committed:
MNT: Re-rendered with conda-smithy 3.1.6 and pinning 2018.07.20

isuruf commented on recipe/meta.yaml:
```diff
@@ -12,42 +12,47 @@ source:
+    - six >=1.10
+    - astropy >=2.0
+    # we require at least numpy 1.14 in pyuvdata
+    - numpy >=1.14
```
Change this to `- numpy 1.14`

isuruf replied:
`- numpy 1.14.*`

isuruf commented on recipe/meta.yaml:
```diff
@@ -12,42 +12,47 @@ source:
+    - six >=1.10
+    - astropy >=2.0
+    # we require at least numpy 1.14 in pyuvdata
+    - numpy >=1.14
```
Change this to `- {{ pin_compatible('numpy') }}`

pkgw replied:
Thanks @isuruf.
